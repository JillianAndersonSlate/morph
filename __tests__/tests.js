import { morph } from '../index.js'
import { join } from 'path'
import { readdirSync, readFileSync } from 'fs'

const as = 'tests'
const isData = f => /\.view.tests$/.test(f)
const getPath = (f = '.') => join(__dirname, 'tests', f)
describe(as, () => {
  readdirSync(getPath())
    .filter(isData)
    .forEach(f => {
      const name = f.replace(/\.view$/, '')
      const raw = getPath(f)
      const code = readFileSync(raw, 'utf-8')

      it(`parses ${as} ${name}`, () => {
        expect(
          morph(code, { as, file: { raw }, name, pretty: true })
        ).toMatchSnapshot()
      })
    })
})
