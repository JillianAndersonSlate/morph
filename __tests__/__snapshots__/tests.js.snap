// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests parses tests MyView.view.tests 1`] = `
Object {
  "code": "export const names = ['Main', 'Next', 'EvenMore', 'Data'];

export const make = display => {
  const Main = {
    width: 360,
    text: \\"it's Main!\\",
    onClick: () => display(Next, 'Next'),
  };
  const Next = {
    width: 360,
    text: \\"it's Next!\\",
    onClick: () => display(EvenMore, 'EvenMore'),
  };
  const EvenMore = {
    width: 360,
    text: \\"it's EvenMore!\\",
    onClick: () => display(Main, 'Main'),
  };
  const Data = Object.assign(
    { width: 360, text: \\"it's Main!\\", onClick: () => display(Next, 'Next') },
    {
      this: 'came from a json file',
    }
  );
  return { _main: 'Main', Main, Next, EvenMore, Data };
};
",
}
`;
